File: Lession 1/InsertionSort.jl
function InsertionSort(A::Vector{Int})
    for i in 2:length(A)
        key = A[i]
        j = i - 1
        while j > 0 && A[j] > key
            A[j + 1] = A[j]
            j = j - 1
        end
        A[j + 1] = key
    end
end

function main()
    A = [5, 2, 4, 6, 1, 3]
    InsertionSort(A)
    println(A)
end

main()

File: Lession 1/InsertionSort.py
def InsertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
    return arr
def main():
    arr = [12, 11, 13, 5, 6]
    print(InsertionSort(arr))
if __name__ == "__main__":
    main()
    